yksgi/src/llvmbridge.rs

```diff
@@ -40,116 +42,140 @@ impl LLVMFunction {
         Self(func)
     }
 
-    pub unsafe fn bb(&self, bbidx: usize) -> LLVMBasicBlock {
-        let mut bb = LLVMGetFirstBasicBlock(self.0);
-        for _ in 0..bbidx {
-            bb = LLVMGetNextBasicBlock(bb);
+    pub fn bb(&self, bbidx: usize) -> LLVMBasicBlock {
+        unsafe {
+            let mut bb = LLVMGetFirstBasicBlock(self.0);
+            for _ in 0..bbidx {
+                bb = LLVMGetNextBasicBlock(bb);
+            }
+            LLVMBasicBlock::new(bb)
         }
-        LLVMBasicBlock::new(bb)
     }
 }
 
-pub struct LLVMBasicBlock(pub LLVMBasicBlockRef);
+pub struct LLVMBasicBlock(LLVMBasicBlockRef);
 
 impl LLVMBasicBlock {
     pub unsafe fn new(bb: LLVMBasicBlockRef) -> Self {
         Self(bb)
     }
 
-    pub unsafe fn first(&self) -> LLVMValue {
+    pub fn unwrap(&self) -> LLVMBasicBlockRef {
+        self.0
+    }
+
+    pub fn first(&self) -> LLVMValue {
         self.instruction(0)
     }
 
-    pub unsafe fn instruction(&self, instridx: usize) -> LLVMValue {
-        let mut instr = LLVMGetFirstInstruction(self.0);
-        for _ in 0..instridx {
-            instr = LLVMGetNextInstruction(instr);
+    pub fn instruction(&self, instridx: usize) -> LLVMValue {
+        unsafe {
```

Comment:
Maybe this is a little anal, but i always try to make unsafe blocks as small as possible.

Here the `for` line needn't be inside unsafe.

Reply:
Agreed, minimising `unsafe` blocks is important.

Reply:
Can do, just means there'll be 3 `unsafe`s instead of one.

Reply:
Given the unsafe was previously in the function, I thought this is equivalent without making the code worse to read.

Reply:
I know, I just thought a for-loop can't possibly be unsafe. But I guess it'll prevent us from adding other unsafe stuff inside the block without noticing. So I'll change it.

Reply:
Fixed in 9a253a4

Reply:
> I just thought a for-loop can't possibly be unsafe.

You can put arbitrary code in the conditional, for example, so yes, it can be unsafe!